### ISSUE TITLE
The ShellcheckMojo Has Parameters Marked as Read-only

### ISSUE THREAD

**Issue by hoshposh** [marco-nicolini/shellcheck-maven-plugin#5]
## Overview

When running a maven build using the `shellcheck-maven-plugin` I see the following warnings:

```console
[WARNING] Parameter 'binaryResolutionMethod' is read-only, must not be used in configuration
[WARNING] Parameter 'skip' (user property 'skip.shellcheck') is read-only, must not be used in configuration
[WARNING] Parameter 'sourceDirs' is read-only, must not be used in configuration
```

When I look in the `ShellcheckMojo` class I see that the majority of arguments for the plugin are marked as `readonly=true`.

Is there an alternative approach to setting the arguments or should the `readonly` flag be relaxed a little?

**Comment by giraciopide** [comment]
Surely they can be relaxed, in fact, reading thoroughly the javadoc for @Parameter annotation having them readonly is wrong.

**Comment by giraciopide** [comment]
In 4015dc7db5983b79183c0cbc86f7907ce9a09a48 I have removed the readonly=true from all parameters in that are meant to be configured directly by users.

I never saw the warnings before because it turns out that it's only maven >= 3.9 that complains about it (rightly so).

**Comment by giraciopide** [comment]
I released 0.5.1 to remove the warnings, will be on maven central shortly.
